### ISSUE TITLE
interface hotkeys configuration

### ISSUE THREAD

**Issue by whismerhill** [endless-sky/endless-sky#2208]
currently interface hotkeys are not tied to ingame hotkeys
of course all hotkeys are disabled while you are typing a ship name or anything

example: despite rebinding up,down,left,right to WSAD, inside the map, showing "ports" with food selected:
-"down arrow" brings me down to next commodity
- "S" brings up the shipyard (whether that is desirable is another subject entirely)

an expansion of the ingame settings, would add the ability to customize interface hotkeys, which are currently hardcoded.
Thoughts ?

**Comment by MessyMix** [comment]
Vague and unclear. What are you asking?

**Comment by KiLEdEnNis** [comment]
@MessyMix like, being able to edit menu hotkeys
like, assign a letter to go to shipyard instead of the default "S"

**Comment by whismerhill** [comment]
an expansion of the ingame settings, would add the ability to customize interface hotkeys, which are currently hardcoded.

was that clearer ???

**Comment by Amazinite** [comment]
Tl;DR: Customizable hotkeys for stuff like the player info screen or when on planets.

**Comment by MessyMix** [comment]
@whismerhill yes, a lot clearer, thanks.

I suppose a separate panel for "user interface" under controls (I wouldn't know where to put this visually, though) would be able to contain these settings. There are, after all, a lot of keys that aren't covered in the current control panel.

**Comment by Jorlem** [comment]
@MessyMix Another upside to this would be as a reference. Currently, there doesn't appear to be a way besides trial and error for new players to learn about the hotkeys that are not on the Preferences screen.

**Comment by KiLEdEnNis** [comment]
@Jorlem hold the Alt key to see what hotkey does what

**Comment by whismerhill** [comment]
I think what's needed is a new button that allows to access all the interface hotkeys if that's ever implemented several choices are possible, all in the preferences menu of the game : 
1 - when clicking on "controls" a submenu shows up (which pushes down "settings" & "plugins" buttons)
this submenu item is labelled "Interface". (Clicking on "controls", still brings up the game general controls)
2 - another button next to the end of the list (above "back to menu") labelled "Interface Controls" this time (because it's separately located from the controls button)
3 - another button just before "back to menu" labelled the same

I think solution 1 is the best but might involve the most work (since it dynamically relocates the other two button.)

**Comment by Jorlem** [comment]
@KiLEdEnNis That only helps if new players are aware of that functionality of the alt key ( which I was not aware of either.)  This is something of a catch 22, as a new player would need to know at least one of the hotkeys that are not documented in-game to learn about the hotkeys that are not documented in-game.

**Comment by endless-sky** [comment]
The UI hotkeys are the equivalent of menu shortcuts. Very few programs (aside from some... *unusual* ones like emacs) let you rebind those shortcuts.
